test/tbb/stub_unsigned_lib.cpp

```diff
@@ -0,0 +1,34 @@
+/*
+    Copyright (c) 2025 Intel Corporation
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+//! \file stub_unsigned_lib.cpp
+//! \brief Test for [internal] functionality
+
+#if _WIN32 || _WIN64
+// Make sure the symbol is visible
+#define STUB_EXPORT __declspec(dllexport)
+#else
+// On other platforms check the branch when symbol cannot be resolved by hiding it
+#define STUB_EXPORT __attribute__ ((visibility ("hidden")))
+#endif
+
+#include <cstdlib>
+
+extern "C" {
+STUB_EXPORT void foo() {
+    std::abort();               // Test dynamic link should never call this function
+}
```

Comment:
Perhaps you could also try to call std::abort during the initialization of the library since it shouldn't ever be loaded.

Reply:
Done.